Neos.ContentRepository.Core/Classes/SharedModel/Exception/AbsoluteNodePathIsInvalid.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\SharedModel\Exception;
+
+/**
+ * The exception to be thrown if an invalid absolute node path was tried to be initialized
+ *
+ * @api because this might have to be handled in the application layer
+ */
+final class AbsoluteNodePathIsInvalid extends \DomainException
```

Comment:
The [explanation for DomainExceptions on php.net](https://www.php.net/manual/en/class.domainexception.php) is

> Exception thrown if a value does not adhere to a defined valid data domain.

or, in German:

> Ausnahme, die ausgelöst wird, wenn ein Wert nicht in einem definierten gültigen Datenbereich liegt.

I don't think that this is the correct type here as "domain" does not refer to DDD domain..

I would suggest to extend `RuntimeException` or `InvalidArgumentException`

Reply:
i actually prefer `\DomainException` for these kind of exceptions, but only because i use them all the time for it ^^

i preserve `RuntimeException` for something more dynamic and unlikely. And `InvalidArgumentException` for cases in php7.4 where union types didnt exist or a string must not be empty in some random method ^^

But this is just preference and i have no data/facts to back this up.